elyra/metadata/metadata_app.py

```diff
@@ -473,6 +473,96 @@ def start(self):
+                        description='Overwrite existing metadata instance with the same name',
+                        default_value=False)
+
+    # 'Import' flags
+    options: List[Option] = [directory_option, replace_flag]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.metadata_manager = MetadataManager(schemaspace=self.schemaspace)
+
+    def start(self):
+        super().start()  # process options
+
+        directory = self.directory_option.value
+
+        src_directory = os.path.join(directory, self.schemaspace)
+
+        try:
+            json_files = [f for f in os.listdir(src_directory) if f.endswith('.json')]
+        except OSError as e:
+            print(f"Unable to reach the '{src_directory}' directory: {e.strerror}: '{e.filename}'")
+            self.exit(1)
+
+        if len(json_files) == 0:
+            print(f"No instances for import found in the '{src_directory}' directory")
+            return
+
+        instance_count_imported = 0
+        instance_count_not_imported = 0
+        metadata_file = None
+
+        for file in json_files:
+            filepath = os.path.join(src_directory, file)
+            try:
+                with open(filepath) as f:
+                    metadata_file = json.loads(f.read())
+                f.close()
+            except OSError as e:
+                print(f"Unable to reach the '{file}' file: {e.strerror}: '{e.filename}'")
+                instance_count_not_imported += 1
+                continue
+
+            name = os.path.splitext(file)[0]
+            try:
+                schema_name = metadata_file['schema_name']
+                display_name = metadata_file['display_name']
+                metadata = metadata_file['metadata']
+            except KeyError as e:
+                print(f"Could not find '{e.args[0]}' key in the import file '{filepath}'")
+                instance_count_not_imported += 1
+                continue
+
+            new_instance = None
+
+            try:
+                if self.replace_flag.value:  # if replacing, fetch the instance so it can be updated
+                    updated_instance = self.metadata_manager.get(name)
+                    updated_instance.schema_name = schema_name
+                    if display_name:
+                        updated_instance.display_name = display_name
+                    if name:
+                        updated_instance.name = name
+                    updated_instance.metadata.update(metadata)
+                    new_instance = self.metadata_manager.update(name, updated_instance)
+                else:  # create a new instance
+                    instance = Metadata(schema_name=schema_name, name=name,
+                                        display_name=display_name, metadata=metadata)
+                    new_instance = self.metadata_manager.create(name, instance)
+            except Exception:
+                pass
```

Comment:
Could we record which `.json` files could not be imported?  Hmm- I suppose you're letting the metadata service logging provide that, which seems fine. 
Would it make sense to collect a list of `file` names that can be tacked onto the log statement for non-imported instances?